relay-server/src/actors/spooler/mod.rs

```diff
@@ -936,6 +956,94 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn dequeue_waits_for_permits() {
+        relay_test::setup();
+        let num_permits = 3;
+        let buffer_guard: Arc<_> = BufferGuard::new(num_permits).into();
+        let config: Arc<_> = Config::from_json_value(serde_json::json!({
```

Comment:
Do we want to make the number of attempts to fetch from disk statically configurable?